imports/ui/signup/signup.jsx

```diff
@@ -70,35 +52,37 @@ export class SignupPage extends Component {
     if (!this.isFormValid()) return
 
     const { info } = this.state
-
     // Copying all the input values and trimming them
     const signupInfo = Object.keys(info).reduce((all, curr) => {
       if (info[curr]) {
-        all[curr] = info[curr].trim()
+        all[curr] = info[curr]
       }
       return all
     }, {})
     const { submitSignupInfo } = actions
     this.props.dispatch(submitSignupInfo(signupInfo))
   }
-  toggleExistingBz = () => {
+
+  handleTermCheckbox = () => {
+    const { info } = this.state
     this.setState({
-      existingBz: !this.state.existingBz
+      info: Object.assign({}, info, {termsAgreement: 'true'})
```

Comment:
"termsAgreement" is not being sent to the server anymore. It doesn't need to be on "info". It should be set on the state directly. It should also have an initial value when the state object is created in the constructor

Reply:
Also, you should use real boolean values, and not 'true' as a string.

Reply:
Also, this code will cause a minor bug that the user can't "un-tick" the checkbox